### ISSUE TITLE
Inline typecasts triggering no-extra-parens

### ISSUE THREAD

**Issue by tomxtobin** [gajus/eslint-plugin-flowtype#181]
An inline typecast that requires parentheses can trigger ESLint's `no-extra-parens` rule, e.g.:
```
(foo: Object).bar()
```
This warns/errors with "Gratuitous parentheses around expression".

**Comment by tengyifei** [comment]
Any idea how to fix this? Would love to contribute.

**Comment by mpyw** [comment]
No updates?

**Comment by gajus** [comment]
This isn't an issue with `eslint-plugin-flowtype`, but an issue with `eslint` `no-extra-parens` rule.

I cannot think of much that this plugin could do to suppress the warning, other than providing its own version of the rule that does not throw the needless warning.

**Comment by romeovs** [comment]
> I cannot think of much that this plugin could do to suppress the warning, other than providing its own version of the rule that does not throw the needless warning.

I agree! So maybe this can be a feature request 😉

**Comment by gajus** [comment]
I am fine with adding a version of `no-extra-parens` to `eslint-plugin-flowtype`. Contributions are welcome.

**Comment by gajus** [comment]
Closing as stale. PR still welcome.
